File: app/src/main/java/com/qimian233/ztool/hook/modules/systemFramework/DisableFlagSecure.java
package com.qimian233.ztool.hook.modules.systemFramework;

import com.qimian233.ztool.hook.base.BaseHookModule;

import android.os.Build;
import io.github.libxposed.api.XposedInterface;
import io.github.libxposed.api.XposedModuleInterface;

import java.lang.reflect.Method;

/**
 * 禁用FLAG_SECURE标志Hook模块
 * 作用：移除安全窗口标志，允许对"安全内容"进行截图
 */
public class DisableFlagSecure extends BaseHookModule {

    public DisableFlagSecure() {}

    @Override
    public String getModuleName() {
        return "disable_flag_secure";
    }

    @Override
    public String[] getTargetPackages() {
        return new String[]{"system"};
    }

    @Override
    public void handleLoadPackage(XposedModuleInterface.PackageLoadedParam param) throws Throwable {
        ClassLoader classLoader = param.getDefaultClassLoader();
        String packageName = param.getPackageName();
        hookAndroidSystem(classLoader);
    }

    private void hookAndroidSystem(ClassLoader classLoader) {
        try {
            log("开始Hook FLAG_SECURE...");
            Class<?> windowStateClass = classLoader.loadClass(
                    "com.android.server.wm.WindowState"
            );

            // Hook isSecureLocked方法，始终返回false
            Method method = windowStateClass.getDeclaredMethod("isSecureLocked");
            this.xposed.hook(method).intercept(chain -> false);

            log("成功Hook WindowState.isSecureLocked()");

        } catch (Throwable t) {
            logError("Hook FLAG_SECURE失败", t);
        }
    }
}

File: app/src/main/java/com/qimian233/ztool/hook/modules/systemframework/DisableFlagSecure.kt
package com.qimian233.ztool.hook.modules.systemframework

import android.annotation.SuppressLint
import com.qimian233.ztool.data.keys.PreferenceKeys
import com.qimian233.ztool.data.keys.ScopeKeys
import com.qimian233.ztool.hook.base.SystemHookModule
import io.github.libxposed.api.XposedModuleInterface.SystemServerStartingParam

/**
 * Disable FLAG_SECURE flag hook module
 * Removes the secure window flag to allow screenshots of "secure" content
 */
@SuppressLint("PrivateApi")
class DisableFlagSecure : SystemHookModule() {
    override fun getModuleName(): String = PreferenceKeys.DISABLE_FLAG_SECURE.name

    override fun getTargetPackages(): Array<String> = arrayOf(ScopeKeys.SYSTEM_SERVER.packageName)

    override fun handleSystemServerStarting(param: SystemServerStartingParam) {
        val classLoader = param.classLoader
        try {
            logger.info("Hooking FLAG_SECURE...")
            val windowStateClass = classLoader.loadClass(
                "com.android.server.wm.WindowState"
            )
            val method = windowStateClass.getDeclaredMethod("isSecureLocked")
            hookWithId(method, "is_secure_locked") { false }
            logger.info("Successfully hooked WindowState.isSecureLocked()")
        } catch (t: Throwable) {
            logger.error("Failed to hook FLAG_SECURE", t)
        }
    }
}
